trio/_core/_multierror.py

```diff
@@ -218,33 +236,54 @@ def __new__(cls, exceptions, *, _collapse=True):
             # In an earlier version of the code, we didn't define __init__ and
             # simply set the `exceptions` attribute directly on the new object.
             # However, linters expect attributes to be initialized in __init__.
+            from_class: type[Self] | type[NonBaseMultiError] = cls
             if all(isinstance(exc, Exception) for exc in exceptions):
-                cls = NonBaseMultiError
+                from_class = NonBaseMultiError
 
-            return super().__new__(cls, "multiple tasks failed", exceptions)
+            # Ignoring arg-type: 'Argument 3 to "__new__" of "BaseExceptionGroup" has incompatible type "list[BaseException]"; expected "Sequence[_BaseExceptionT_co]"'
+            # We have checked that exceptions is indeed a list of BaseException objects, this is fine.
+            new_obj = super().__new__(from_class, "multiple tasks failed", exceptions)  # type: ignore[arg-type]
+            assert isinstance(new_obj, (cls, NonBaseMultiError))
+            return new_obj
 
-    def __reduce__(self):
+    def __reduce__(
+        self,
+    ) -> tuple[object, tuple[type[Self], list[BaseException]], dict[str, bool],]:
```

Comment:
why the trailing comma?

Reply:
I used black formatting, not sure, I can try removing it.